Content.Client/Chat/ChatManager.cs

```diff
@@ -104,5 +145,81 @@ private void _onChatBoxTextSubmitted(ChatBox chatBox, string text)
                 }
             }
         }
+
+        private void _onFilterButtonToggled(ChatBox chatBox, Button.ButtonEventArgs e)
+        {
+            switch (e.Button.Name)
+            {
+                case "OOC":
+
+                _OOCstate = !_OOCstate;
```

Comment:
Why are you storing these? Can't you simply get the button state from the event?